cmd/action/delete/ebs/volume/runner.go

```diff
@@ -0,0 +1,114 @@
+package volume
+
+import (
+	"context"
+
+	"github.com/giantswarm/k8sclient/v5/pkg/k8sclient"
+	"github.com/giantswarm/microerror"
+	"github.com/giantswarm/micrologger"
+	"github.com/spf13/cobra"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/kubernetes"
+
+	"github.com/giantswarm/awscnfm/v15/pkg/client"
+	"github.com/giantswarm/awscnfm/v15/pkg/env"
+	"github.com/giantswarm/awscnfm/v15/pkg/key"
+)
+
+type runner struct {
+	flag   *flag
+	logger micrologger.Logger
+}
+
+func (r *runner) Run(cmd *cobra.Command, args []string) error {
+	ctx := context.Background()
+
+	err := r.flag.Validate()
+	if err != nil {
+		return microerror.Mask(err)
+	}
+
+	err = r.run(ctx, cmd, args)
+	if err != nil {
+		return microerror.Mask(err)
+	}
+
+	return nil
+}
+
+func (r *runner) run(ctx context.Context, cmd *cobra.Command, args []string) error {
+	var err error
+
+	var cpClients k8sclient.Interface
+	{
+		c := client.ControlPlaneConfig{
+			Logger: r.logger,
+
+			KubeConfig: env.ControlPlaneKubeConfig(),
+		}
+
+		cpClients, err = client.NewControlPlane(c)
+		if err != nil {
+			return microerror.Mask(err)
+		}
+	}
+
+	var tcClients k8sclient.Interface
+	{
+		c := client.TenantClusterConfig{
+			ControlPlane: cpClients,
+			Logger:       r.logger,
+
+			TenantCluster: r.flag.TenantCluster,
+		}
+
+		tcClients, err = client.NewTenantCluster(c)
+		if err != nil {
+			return microerror.Mask(err)
+		}
+	}
+
+	err = r.cleanupEBSTestResources(ctx, tcClients.K8sClient())
+	if err != nil {
+		return microerror.Mask(err)
+	}
+
+	return nil
+}
+
+// cleanupEBSTestResources will cleanup EBS volume test resources
+func (r *runner) cleanupEBSTestResources(ctx context.Context, tcClient kubernetes.Interface) error {
+	err := tcClient.BatchV1().Jobs("default").Delete(ctx, key.EBSTestJobName(), metav1.DeleteOptions{})
+	if err != nil {
+		return microerror.Mask(err)
+	}
+
+	err = tcClient.RbacV1().ClusterRoles().Delete(ctx, "enable-ebs-psp", metav1.DeleteOptions{})
+	if err != nil {
+		return microerror.Mask(err)
+	}
+
+	err = tcClient.RbacV1().RoleBindings("default").Delete(ctx, "ebs-rolebinding", metav1.DeleteOptions{})
+	if err != nil {
+		return microerror.Mask(err)
+	}
+
+	err = tcClient.CoreV1().PersistentVolumeClaims("default").Delete(ctx, "ebs-claim", metav1.DeleteOptions{})
+	if err != nil {
+		return microerror.Mask(err)
+	}
+
+	pods, err := tcClient.CoreV1().Pods("default").List(ctx, metav1.ListOptions{
+		LabelSelector: "app=awscnfm-ebs-volume-test",
+	})
+	if err != nil {
+		return microerror.Mask(err)
+	}
+
+	err = tcClient.CoreV1().Pods("default").Delete(ctx, pods.Items[0].GetName(), metav1.DeleteOptions{})
+	if err != nil {
+		return microerror.Mask(err)
+	}
+
+	return nil
+}
```

Comment:
one minor thing, if somehow some of the resources are missing then it will exit on an error and won't try to delete the remaining resources, so maybe changing the cod os it will always try to delete all resources and then maybe print an array of errors if there were any?

Reply:
good point, I'll change that 👍🏻